tests/test_cli_compile.py

```diff
@@ -839,3 +839,27 @@ def test_empty_input_file_no_header(runner, empty_input_pkg, prior_output_pkg):
 
     with open("requirements.txt", "r") as req_txt:
         assert req_txt.read().strip() == ""
+
+
+def test_pip_compile_upgrade_package(pip_conf, runner):
+    """
+    Tests pip-compile --upgrade-package shouldn't remove "via" annotation.
+    Issue #929
```

Comment:
```suggestion
    See: GH-929
```
Better to use a project-specific issue marker to avoid issue trackers collision in a possible future.